core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchActivity.java

```diff
@@ -79,20 +79,19 @@ protected void onCreate(Bundle savedInstanceState) {
     if (savedInstanceState != null) {
       searchText = savedInstanceState.getString(EXTRA_SEARCH_TEXT);
     }
+
     Toolbar toolbar = findViewById(R.id.toolbar);
     ViewCompat.setLayoutDirection(toolbar, ViewCompat.LAYOUT_DIRECTION_LOCALE);
     setSupportActionBar(toolbar);
     getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_action_back);
     getSupportActionBar().setHomeButtonEnabled(true);
     searchPresenter.attachView(this);
-    listView = findViewById(R.id.search_list);
+    recyclerView = findViewById(R.id.search_list);
```

Comment:
butterknife

Reply:
Using butterknife on Toolbar is showing an error and won't compile.
```
@BindView(R.id.toolbar) Toolbar toolbar;
```
It says that: the attribute must be constant. On compiling it says that bindview can not be done on toolbar.

Reply:
The problem is that we are in a library module when in `core`.

Did you google `butterknife the attribute must be constant.`? It would have lead to [this issue](https://github.com/JakeWharton/butterknife/issues/468) which then links to [the solution](https://github.com/JakeWharton/butterknife#library-projects) which is the readme on github.

tl;dr use R2, instead of R

Reply:
I did but I only checked some StackOverflow's answers which did not solve the problem. I never knew that when working on core, it is a library module.

Reply:
There is one more
```
    Toolbar toolbar = findViewById(R.id.toolbar);
    setSupportActionBar(toolbar);
```
I think the same toolbar variable can be again used here, I feel like I can get rid of this variable. As I did not did this, do it serve any purpose? I think no.

Reply:
Use butterknife to bind any views (when using java) and if you see `findViewById` get rid of it please